src/DelegationManagement.sol

```diff
@@ -65,19 +66,14 @@ contract DelegationManagementContract {
     mapping(bytes32 => bool) public collectionLock;
     mapping(bytes32 => bool) public collectionUsecaseLock;
 
-    // Constructor
-    constructor() {
-        useCaseCounter = 17;
-    }
-
     /**
      * @notice Delegator assigns a delegation address for a specific use case on a specific NFT collection for a certain duration
      * @notice _collectionAddress --> ALL_COLLECTIONS = Applies to all collections
      * @notice For all Tokens-- > _allTokens needs to be true, _tokenId does not matter
      */
 
     function registerDelegationAddress(address _collectionAddress, address _delegationAddress, uint256 _expiryDate, uint8 _useCase, bool _allTokens, uint256 _tokenId) public {
-        require((_useCase > 0 && _useCase < useCaseCounter) || (_useCase == USE_CASE_CONSOLIDATION));
+        require((_useCase > 0 && _useCase < USE_CASE_ALL_COUNTER) || (_useCase == USE_CASE_CONSOLIDATION) || (_useCase == USE_CASE_SUB_DELEGATION));
```

Comment:
Shouldn't it be like this:

`uint8 constant USE_CASE_ALL_COUNTER = 15;
`

making this line:
`require((_useCase > 0 && _useCase <= USE_CASE_ALL_COUNTER) ||  ...`

Reply:
It was previously `17`, which was one more than the highest use-case id, so i was following that same pattern. 

I think your suggestion here actually makes much more sense. There is 15 use-cases, that should be the number we are tracking. I'll incorporate this.